File: pajton/python/horner.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

def horner(stopien, tabwsp, x):
    wynik = tabwsp[0]
    for i in range(1,4):
        wynik = wynik*x + tabwsp[i]
    return wynik


def main(args):
    stopien = 3
    x = float(raw_input("Podaj x: "))
    tabwsp=[] #pusta lista
    for i in range(4):
        w=float(raw_input("wspolczynnik %s: " %i))
        tabwsp.append(w)
    print x, tabwsp
    print "wynik: ", horner(stopien, tabwsp, x)
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: gittest/c++/cpp/horner.cpp
/*
 * hornen-it
 *
 */


#include <iostream>

using namespace std;

float horner_it(int stopien, int tabwsp[], float x){
    float wynik=tabwsp[0];

    for(int i=1; i <=stopien; i++){
        wynik=wynik*x +tabwsp[i];
        }
    return wynik;
}

int main(int argc, char **argv)

{
    int stopien=3;
    int tabwsp[stopien+1];
    float x=0;
    for(int i=0; i <= stopien; i++){
        cout<<"podaj liczbe "<<i+1<<": ";
        cin>>tabwsp[i];
        }

    cout << "podaj x " << endl;
    cin >> x;
    cout<<"wielomian " << endl;
    cout << horner_it(stopien, tabwsp, x) << endl;


	return 0;
}

